csvcubed/csvcubed/schema/cube-config-schema.jsonc

```diff
@@ -0,0 +1,303 @@
+{
+	"$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "http://gss-cogs.github.io/gss-cogs/csvcubed/csvcubed/schemas/cube-config-schema.jsonc",
+    "title": "Schema for a qb-config cube",
+    "description": "Schema to be used for creation & validation of a single csv-cubed-flavoured linked data cube in CSV-W format",
+    "type": "object",
+
+    "properties": {
+
+		"publisher": {
+            "description": "The schema publisher link uri",
+            "type": "string",
+            "format": "uri"
+		},
+
+        "creator": {
+            "description": "The schema publisher link uri",
+            "type": "string",
+            "format": "uri"
+        },
+
+        "spatial_bound": {
+            "description": "URI that defines the spatial / geographic bounds of the data contained herein",
+            "type": "string",
+            "format": "uri",
+            "default": "https://tdg.docbook.org/tdg/publishers/5.1/dcterms.html#dcterms.spatial"
+        },
+
+        "temporal_bound": {
+            "description": "URI that defines the temporal bounds of the data contained herein",
+            "type": "string",
+            "format": "uri",
+            "default": "https://tdg.docbook.org/tdg/publishers/5.1/dcterms.html#dcterms.temporal"
+        },
+
+        "columns": {
+            "description": "Columns in the same order as the csv, unspecified column types are assumed to be dimensions, case insensitive",
+			"properties": {
+				"Existing dimension": {"$ref": "#/$defs/existingDimension"},
+				"New dimension": {"$ref": "#/$defs/newDimension"},
+				"Existing attribute": {"$ref": "#/$defs/existingAttribute"},
+				"New attribute": {"$ref": "#/$defs/newAttribute"},
+				"Observation": {"$ref": "#/$defs/observation"},
+				"measures": {"$ref": "#/$defs/measure"},
+				"list of units": {"$ref": "#/$defs/unit"}
+			}
+        }
+    },
+
+    "$defs": {
+		"unitValueTypes": {
+			"type": "object",
+			"oneOf": [
+				{"type": "boolean"},
+				{"Existing unit value from unit column": {"$ref": "#/$defs/unit" }},
+				{"New unit value from unit column": {"$ref": "#/$defs/unit" }}
+			]
+		},
+
+		"unit": {
+			"description": "Definition of units for the observed value",
+			"type": "object",
+			"properties": {
+				"label": {
+					"description": "Replacement label for the Unit",
+					"type": "string"
+				},
+				"from_existing": {
+					"description": "URI of an Existing Qb Dimension",
```

Comment:
The `from_existing` properties should provide users with a selection of pre-defined values they can select from as well as allowing them to override that selection with their own string value. 

See [here](https://github.com/GSS-Cogs/family-schemas/blob/main/dataset-schema-1.1.0.json#L820) for an example of how that works in our existing schemas. This definition enables vscode to provide a list of suitable values to select from.

The same behaviour should happen for all other `from_existing` properties, but they should be pointing at the relevant URIs for that *type*.

Reply:
Could you explain this one a little further. We're not currently proposing to fetch the existing values from the defined component? My understanding is that would have to be done when we load the qube_config into csvcubed?

Reply:
I want this: 

![image](https://user-images.githubusercontent.com/73846016/151210544-7ca10710-dea4-40ce-96d0-7384f34ab8e9.png)

Reply:
Admittedly, I suspect our list will be a lot shorter, and depends on @rossbowen 's work in #285

Reply:
I've updated the 'from_existing' attributes to refer to ($ref") the family schemas http://gss-cogs.github.io/family-schemas/external.json#/definitions/...